conda/recipes/cuml/meta.yaml

```diff
@@ -95,6 +95,8 @@ requirements:
     - rapids-dask-dependency ={{ minor_version }}
     - treelite {{ treelite_version }}
     - rapids-logger =0.1
+  run_constrained:
+    - scikit-learn >=1.5.0,<2.0a0
```

Comment:
For the `cuml.accel` deps, we may want to also add `umap` and `hdbscan` here too. This would improve the install story for 0cc for those packages via conda, but wouldn't for `pip`.

An alternate solution is a meta package for conda (`cuml-accel`, or more flexibly `cuml-umap`/`cuml-hdbscan` so we don't require all 3 deps if a user is only using one of them) and extras for `pip` (`cuml[accel]`/`cuml[umap]`/`cuml[hdbscan]`).

I have a _slight_ preference for the `run_constrained` solution for conda, as it composes better with existing workloads (and doesn't lead to an explosion of meta packages for each optional dep). Using `extras` for `pip` and `run_constrained` for conda seems fine to me.

Reply:
I share your preference for `run_constrained`.